### PR TITLE
hook up create/join session

### PR DESCRIPTION
**Pull request by vickyteng**

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  
+    } else {
+      sessions.users.push({
+        name: name,
+        role: userRole
+      });
+
+      sessions.save(function(err) {
+        console.log(`Adding user to session...`)
+        if (err) if (err) console.log(`Error adding user to session: ${err}`);
```
Comment (aigyptios) [inline server/app.js]: You have an extra `if (err)`

File: src/actions/sessionActions.js [hunk 2]
```diff
@@ -19,21 +19,95 @@ export const startStartSession = name => (
         });
     })
 );
+*/
+
+export function startStartSession (name) {
+    return function (dispatch) {
+        fetch('http://localhost:3005/api/create-session',  {
```
Comment (juanflo) [inline src/actions/sessionActions.js]: Rather than hardcode this you could use a .env file

File: server/app.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
+require('dotenv').config();
+var express = require('express');
+var path = require('path');
+//var cookieParser = require('cookie-parser');
```
Comment (aigyptios) [inline server/app.js]: Could you remove all the commented-out code in this file?

File: server/app.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  Session.find({}, function(err, sessions) {
+    if (err) console.log(`Error getting users: ${err}`);;
+
+    return res.send(sessions);
+  })
+})
+
+// Returns session ID and user ID (client wants both)
+app.post('/api/create-session', function(req, res) { 
+  console.log("Creating session..");
```
Comment (aigyptios) [inline server/app.js]: If the code works, I don't know whether we need any `console.log`s anymore (throughout the file).

File: server/models/session.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+var mongoose = require('mongoose');
+
+/*
```
Comment (aigyptios) [inline server/models/session.js]: Remove unnecessary comments.

**Comment by juanflo** [comment]
@vickyteng I would remove the server code since there is a separate repo for that.

File: src/actions/sessionActions.js [hunk 6]
```diff
@@ -7,7 +7,7 @@ export const startSession = (session, userId, moderator) => ({
     userId,
     moderator
 });
-
+/*
```
Comment (aigyptios) [inline src/actions/sessionActions.js]: Remove commented-out code.

File: src/actions/sessionActions.js [hunk 7]
```diff
@@ -19,21 +19,95 @@ export const startStartSession = name => (
...
+            console.log('Request failure: ', err)
+        })
+    }
+    
+    return updateDatabase;
+}
 
-export const startJoinSession = (session, user) => {
+
+/*export const startJoinSession = (session, user) => {
```
Comment (aigyptios) [inline src/actions/sessionActions.js]: Remove commented-out code.

File: src/actions/sessionActions.js [hunk 8]
```diff
@@ -19,21 +19,95 @@ export const startStartSession = name => (
...
+}
+
+export function startJoinSession (session, user) {
+    let updateDatabase;
+    const {id: userId, ...userInfo} = user;
+
+    // if moderator    
+    if (userId) {
+        updateDatabase = dispatch =>
+        fetch(`http://localhost:3005/api/join-session/${session}`, {
```
Comment (juanflo) [inline src/actions/sessionActions.js]: Same as above

File: src/actions/sessionActions.js (hunk 2 above)
Comment (sinkng) [reply src/actions/sessionActions.js]: I would create api.js and move all the fetch to that file, so when changing the url it's tough to find and replace all.

File: src/actions/sessionActions.js (hunk 8 above)
Comment (sinkng) [reply src/actions/sessionActions.js]: I would create api.js and move all the fetch to that file, so when changing the url it's tough to find and replace all.

**Pull request merged**